### PR TITLE
Keep track of diff_to_superseded parameter in all the cases

### PR DESCRIPTION
**Pull request by ncounter**
Fix https://github.com/openSUSE/open-build-service/issues/15916

# Test:

### tl;dr:
As long as the review-app is not re-genarated, the following steps are already reflected into https://obs-reviewlab.opensuse.org/ncounter-fix-diff-to-superseded-changes/request/show/19/changes?diff_to_superseded=18

### long version
1. Go https://obs-reviewlab.opensuse.org/ncounter-fix-diff-to-superseded-changes/package/show/home:Admin/hello_world
2. Branch the package
3. Change some lines in README.txt and save
4. Submit the package (it creates a new submit request)
5. Go back to the branched package and change few more lines in the README.txt again
6. Submit the package again and checking that the submit request supersedes the previous one (a flag to be marked while creating the submit request)
7. Visit the latest submit request page at the changes tab (if in beta), it will show the full diff of the package from the initial branched one
8. Visit now the _(changes from superseded)_ link, it's in the header of the submit request
9. The visible diff includes only the changes from the first submit request to the latest one

### CODE REVIEW

**Comment by ncounter** [comment]
> Just so this doesn't get merged until the changes to diff comments are removed, since they are unrelated to the issue

Done, that change is dropped.

**Review by saraycp — APPROVED** [review]
The diff issue is solved, you test it out here:

https://obs-reviewlab.opensuse.org/ncounter-fix-diff-to-superseded-changes/request/show/19

I'll take care of the test coverage in a follow-up PR so the fix is deployed soonish.

**Comment by saraycp** [comment]
My last force-push is wrong, sorry :disappointed: I leave it in draft until @ncounter pushes the correct version again

**Comment by ncounter** [comment]
> My last force-push is wrong, sorry 😞 I leave it in draft until @ncounter pushes the correct version again

No problem :slightly_smiling_face: . I pushed back the final version.

**Comment by ncounter** [comment]
Rebased on master (few changes happened meanwhile) for the last round of tests. If green, I'll move forward merging this.
